src/Streamly/Coreutils/Uniq.hs

```diff
@@ -0,0 +1,161 @@
+import qualified Streamly.Memory.Array as A
+import qualified Streamly.Internal.Data.Fold as FL
+import qualified Streamly.Data.Unicode.Stream as Un
+import qualified Streamly.Internal.FileSystem.File as File
+
+import Data.Word (Word8)
+import Data.Char (toLower)
+import System.IO (Handle, stdout)
+import System.IO.Unsafe (unsafePerformIO)
+import Foreign.Storable (Storable)
+import Control.Monad.IO.Class (MonadIO)
+import Streamly.Internal.Data.Stream.StreamK.Type (IsStream)
+
+-------------------------------------------------------------------------------
+-- Record for options used with uniq
+-------------------------------------------------------------------------------
+
+{-# INLINE intMax #-}
+
+intMax :: Int
+intMax = maxBound
+
+
+data UniqOptions = UniqOptions {
+                       count :: Bool
+                     , repeated :: Bool       -- display only duplicate lines once for each group
+                     , duplicate :: Bool      -- print all duplicate lines
+                     , skipFields :: Int      -- skips fields, then chars & checks remaining
+                     , skipChar :: Int
+                     , checkChar :: Int
+                     , ignoreCase :: Bool
+                     , unique :: Bool         -- print only unique lines
+                     , zeroTerminated :: Bool
+                  }
+
+defaultUniqOptions :: UniqOptions
+defaultUniqOptions = UniqOptions True False False 0 0 intMax False True False
+
+
+-------------------------------------------------------------------------------
+-- helper functions for uniq
+-------------------------------------------------------------------------------
+
+
+-- convert stream of bytes to stream of characters - ignore case if bool is True
+
+ignCase :: (IsStream t, Monad m) => Bool -> t m Word8 -> t m Char
+ignCase flag strm = if flag == True
+                    then S.map toLower $ Un.decodeLatin1 strm
+                    else Un.decodeLatin1 strm
+
+
+-- convert stream of char to stream of Array Char by breaking on newline
+
+splitOnNewLine :: (MonadIO m, IsStream t, Monad m) => t m Char -> t m (A.Array Char)
+splitOnNewLine strm = S.splitOnSuffix (== '\n') A.write strm
+
+
+isSpace :: Char -> Bool
+isSpace c = c == ' '
+
+safeLast :: [a] -> Maybe a
+safeLast [] = Nothing
+safeLast (x:[]) = Just x
+safeLast (x:xl) = safeLast xl
+
+
+mergeStringChar :: String -> Char -> String
+mergeStringChar s c = case safeLast s of
+                                  Just l -> if isSpace l == True && isSpace c == True
+                                             then s ++ [c]
+                                             else if isSpace l == False && isSpace c == False
+                                             then s ++ [c]
+                                             else [c]
+                                  Nothing -> s ++ [c]
+
+
+
+skpField :: (IsStream t, Monad m) => Int -> A.Array Char -> t m Char
+skpField n arr = S.concatMap (\s -> S.fromList s)
```

Comment:
Is this incomplete?

Reply:
Also, what do you expect the output of `skpField _ (['a', 'b', 'c', 'd'] :: Array Char)` to be?

Reply:
Yes, this was incomplete.